src/NetscriptFunctions/Corporation.ts

```diff
@@ -309,6 +310,7 @@ export function NetscriptCorporation(
       upgrades: division.upgrades.slice(),
       cities: cities,
       products: division.products === undefined ? [] : Object.keys(division.products),
+      makesProducts: getDivision(division.name).makesProducts,
```

Comment:
we shouldn't need to get the division again 

```suggestion
      makesProducts: division.makesProducts,
```

should suffice

Reply:
Its a different type of division. `IIndustry` vs `NSDivision`